### PR TITLE
[WIP] This significantly reduces steady state malloc's on steady state. PR only for testing.

### PR DESCRIPTION
**Pull request by kumpera**

### CODE REVIEW

File: mono/metadata/class.c [hunk 1]
```diff
@@ -82,6 +82,74 @@ static void mono_generic_class_setup_parent (MonoClass *klass, MonoClass *gklass
...
+//128 bytes on 64bits
+#define MAX_ADAPT_PAIRS 7
+
+typedef struct {
+	GHashTable *overflow;
+	int count;
+	gpointer keys [MAX_ADAPT_PAIRS];
+	gpointer values [MAX_ADAPT_PAIRS];
+} MonoAdaptatativeHashTable;
+
```
Comment (vargaz) [inline mono/metadata/class.c]: Not sure that Adaptatative is a word.

Comment (Therzok) [inline mono/metadata/class.c]: Should it be adaptive?

Comment (evincarofautumn) [inline mono/metadata/class.c]: Yup, adaptive. Technically they’re synonyms, but adaptative is uncommon/old-fashioned.

**Comment by kumpera** [comment]
build

**Comment by kumpera** [comment]
build

**Comment by kumpera** [comment]
The last commit fixes the windows build

**Pull request closed without merging**